graphql/typeDefs.ts

```diff
@@ -57,6 +58,22 @@ export default gql`
     deleteComment(id: Int!): Comment
     addModule(lessonId: Int!, name: String!, content: String!): Module
     deleteModule(id: Int!): SuccessResponse
+    addExercise(
+      moduleId: Int!
+      description: String!
+      answer: String!
+      testable: Boolean!
+      testStr: String
+    ): Exercise
```

Comment:
@mino323 Question. Why is it for `addExercise` we return type `Exercise`, but for ` updateExercise`, we return `Exercise!`?(One uses exclamation point, the other doesn't).

Reply:
My thinking was when you add an exercise you might not need anything returned. When you update you want to see the updated data.

Reply:
My thinking is whenever you create/update/delete a resource, it should return that resource. Not sure though

Reply:
I agree with Karla. In my point of view, it works as a confirmation that this request is successful.

Reply:
@SahilKalra98 is correct. whenever you create/update a resource, unless there is error you should be returning the resource. I'm not sure about delete though, my memory tells me you simply return the id of the deleted resource (no point returning a data that no longer exists).

Reply:
Got it thanks, all!